File: job-portal-frontend/src/api/userService.js
import apiClient from './api';

export const getAllUsers = () => {
    return apiClient.get('/api/users');
};

export const getUserById = (id) => {
    return apiClient.get(`/api/users/${id}`);
};

export const createUser = (userData) => {
    return apiClient.post('/api/users', userData);
};

export const updateUser = (id, userData) => {
    return apiClient.put(`/api/users/${id}`, userData);
};

export const deleteUser = (id) => {
    return apiClient.delete(`/api/users/${id}`);
};

File: job-portal-backend/userservice/src/main/java/com/carrerconnect/userservice/service/UserService.java
package com.carrerconnect.userservice.service;

import java.util.List;

import com.carrerconnect.userservice.dto.UserDTO;
import com.carrerconnect.userservice.exception.CareerConnectException;


public interface UserService {
	 UserDTO createUser(UserDTO userDTO);
	    UserDTO getUserById(Long id) throws CareerConnectException;
	    List<UserDTO> getAllUsers();
	    UserDTO updateUser(Long id, UserDTO userDTO);
	    void deleteUser(Long id) throws CareerConnectException;
}
